metis-indexing/src/test/java/eu/europeana/indexing/solr/property/FullBeanSolrPropertiesTest.java

```diff
@@ -0,0 +1,88 @@
+package eu.europeana.indexing.solr.property;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import eu.europeana.corelib.definitions.edm.entity.Place;
+import eu.europeana.corelib.definitions.solr.DocType;
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.corelib.solr.entity.AgentImpl;
+import eu.europeana.corelib.solr.entity.PlaceImpl;
+import eu.europeana.corelib.solr.entity.ProxyImpl;
+import java.sql.Date;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import org.apache.solr.common.SolrInputDocument;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FullBeanSolrPropertiesTest {
+
+
+  private FullBeanSolrProperties fullBeanSolrProperties;
+  private SolrInputDocument solrInputDocument;
+  private FullBeanImpl fullBean;
+
+
+  @BeforeEach
+  void setup() {
+    solrInputDocument = new SolrInputDocument();
+    fullBean = new FullBeanImpl();
+    fullBeanSolrProperties = new FullBeanSolrProperties();
+  }
+
+
+  @Test
+  void fullBeanSolrSetPropertiesTest() {
+
+    // proxy setup
+    ProxyImpl proxy = new ProxyImpl();
+    proxy.setAbout("About Proxy");
+    proxy.setEdmType(DocType.TEXT.getEnumNameValue());
+    proxy.setEdmHasType(Map.of("edm types", List.of(DocType.TEXT.getEnumNameValue())));
+    proxy.setEuropeanaProxy(true);
+    proxy.setEdmCurrentLocation(Map.of("Somewhere", List.of("Nowhere")));
+    proxy.setDctermsSpatial(Map.of("Somewhere", List.of("Nowhere")));
+
+    fullBean.setProxies(List.of(proxy));
```

Comment:
Add a new test with the following definition of proxies.
```
    List<ProxyImpl> proxies = new ArrayList<>();
    proxies.add(proxy);
    proxies.add(proxy);
    proxies.add(null);
```
to cover this line
eu/europeana/indexing/solr/property/FullBeanSolrProperties.java:48
```
    final String[] types = Optional.ofNullable(fullBean.getProxies()).stream().flatMap(List::stream)
                                   .filter(Objects::nonNull).map(ProxyImpl::getEdmType).filter(Objects::nonNull).distinct()
                                   .toArray(String[]::new);
```

Reply:
This setup throws NPE. Fix is in FullBeanSolrProperties.java:69